packages/protocol/src/certificate/CertificateAuthority.ts

```diff
@@ -48,49 +65,64 @@ export class CertificateAuthority {
         return this.#construction;
     }
 
-    static async create(crypto: Crypto, options?: StorageContext | CertificateAuthority.Configuration) {
+    static async create(
+        crypto: Crypto,
+        options?: StorageContext | CertificateAuthority.Configuration | CertificateAuthority.PartialConfiguration,
+    ) {
         return asyncNew(CertificateAuthority, crypto, options);
     }
 
-    constructor(crypto: Crypto, options?: StorageContext | CertificateAuthority.Configuration) {
+    constructor(
+        crypto: Crypto,
+        options?: StorageContext | CertificateAuthority.Configuration | CertificateAuthority.PartialConfiguration,
+    ) {
         this.#crypto = crypto;
+
         this.#construction = Construction(this, async () => {
-            // Use provided CA config or read from storage, otherwise initialize and store
             const certValues = options instanceof StorageContext ? await options.values() : (options ?? {});
 
+            if (this.#isValidStoredRootCertificate(certValues)) {
+                const hasIcacInStorage = this.#isValidStoredIcacCertificate(certValues);
+                if (hasIcacInStorage) {
+                    this.#intermediateCert = true;
+                } else if (typeof certValues.intermediateCert === "boolean") {
+                    this.#intermediateCert = certValues.intermediateCert;
+                } else {
+                    this.#intermediateCert = false;
+                }
```

Comment:
I think it'd be best to throw an ImplementationError here if intermediateCert from storage is true but ICAC props are missing since in this code path we're not reading from config